src/transition.jl

```diff
@@ -0,0 +1,17 @@
+abstract type AbstractTransition end
+
+struct IndependentMomentumTransition <: AbstractTransition
+    h::AbstractSystem
+end
```

Comment:
```suggestion
struct IndependentMomentumTransition{S<:AbstractSystem} <: AbstractTransition
    h::S
end
```

Same logic as suggestion above about avoiding abstractly typed fields and using a parametric type instead.